### ISSUE TITLE
Failure to create cache key on invalid symlink

### ISSUE THREAD

**Issue by kdeldycke** [astral-sh/ruff#19421]
### Summary

Tl;Dr: an invalid symlink makes `ruff` exits with code 1 due to cache key creation failure.

### How to reproduce

```shell-session
$ ln -s /blah.py

$ ls -lah
Permissions Size User Date Modified    Name
drwxr-xr-x@    - kde  2025-07-18 15:30 .
drwxr-xr-x     - kde  2025-07-18 15:30 ..
drwxr-xr-x@    - kde  2025-07-18 15:30 .ruff_cache
lrwxr-xr-x     - kde  2025-07-18 15:30 blah.py -> /blah.py

$ cat /blah.py                                                                
cat: /blah.py: No such file or directory (os error 2)

$ ruff check       
blah.py:1:1: E902 Failed to create cache key
  Cause: Failed to create cache key
  Cause: No such file or directory (os error 2)
Found 1 error.

$ echo $?        
1

$ rm ./blah.py

$ ruff check       
warning: No Python files found under the given path(s)
All checks passed!

$ echo $?      
0
```

### Expected behavior

I expect the `ruff` CLI to emit a warning and skip the invalid symlink altogether, and continue on checking other files.

An alternative would be to allow tweaking this behavior, i.e. either silently ignoring the broken symlink or erroring on purpose, behind an option.

### Version

ruff 0.12.3

**Comment by kdeldycke** [comment]
Might be related to https://github.com/astral-sh/ruff/issues/19395
